bigquery_etl/cli/query.py

```diff
@@ -768,73 +769,166 @@ def update(name, sql_dir, project_id):
-        project_name = query_file_path.parent.parent.parent.name
+        if update_downstream:
+            # update downstream dependencies
+            if changed:
+                if not is_authenticated():
+                    click.echo(
+                        "Cannot update downstream dependencies."
+                        "Authentication to GCP required. Run `gcloud auth login` "
+                        "and check that the project is set correctly."
+                    )
+                    sys.exit(1)
 
-        # update bigquery metadata
-        try:
-            client = bigquery.Client()
-            table = client.get_table(f"{project_name}.{dataset_name}.{table_name}")
-            metadata = Metadata.of_query_file(str(query_file_path))
-            metadata_file_path = query_file_path.parent / METADATA_FILE
-
-            if table.time_partitioning and (
-                metadata.bigquery is None or metadata.bigquery.time_partitioning is None
-            ):
-                metadata.set_bigquery_partitioning(
-                    field=table.time_partitioning.field,
-                    partition_type=table.time_partitioning.type_.lower(),
-                    required=table.time_partitioning.require_partition_filter,
+                table = query_file.parent.name
+                dataset = query_file.parent.parent.name
+                project = query_file.parent.parent.parent.name
+                identifier = f"{project}.{dataset}.{table}"
+                tmp_identifier = f"{project}.{tmp_dataset}.{table}_" + "".join(
+                    random.choice(string.ascii_lowercase) for i in range(12)
                 )
-                click.echo(f"Partitioning metadata added to {metadata_file_path}")
 
-            if table.clustering_fields and (
-                metadata.bigquery is None or metadata.bigquery.clustering is None
-            ):
-                metadata.set_bigquery_clustering(table.clustering_fields)
-                click.echo(f"Clustering metadata added to {metadata_file_path}")
+                # create temporary table with updated schema
+                client = bigquery.Client()
 
-            metadata.write(metadata_file_path)
-        except NotFound:
-            click.echo(
-                f"Destination table {project_name}.{dataset_name}.{table_name} "
-                "does not exist in BigQuery. Run bqetl query schema deploy "
-                "<dataset>.<table> to create the destination table."
-            )
+                tmp_schema_file = NamedTemporaryFile()
+                schema = Schema.from_schema_file(query_file.parent / SCHEMA_FILE)
+                schema.to_json_file(Path(tmp_schema_file.name))
+                bigquery_schema = client.schema_from_json(tmp_schema_file.name)
+                tmp_table = bigquery.Table(tmp_identifier, schema=bigquery_schema)
+                client.create_table(tmp_table)
+
+                tmp_tables[identifier] = tmp_identifier
```

Comment:
Create a temporary table with the updated schema and use this temporary table in all downstream dependencies